skopt/learning/gaussian_process/tests/test_kernels.py

```diff
@@ -69,12 +70,12 @@ def check_gradient_correctness(kernel, X, Y, step_size=1e-4):
     assert_array_almost_equal(X_grad, num_grad, decimal=3)
 
 
-def test_gradient_correctness():
+@pytest.mark.parametrize("kernel", KERNELS)
+def test_gradient_correctness(kernel):
     rng = np.random.RandomState(0)
     X = rng.randn(5)
     Y = rng.randn(10, 5)
-    for kernel in KERNELS:
-        yield check_gradient_correctness, kernel, X, Y
+    check_gradient_correctness(kernel, X, Y)
 
 
 def test_gradient_finiteness():
```

Comment:
Ah yes, I missed that one. That test would become:
```
@pytest.mark.parametrize("random_state, kernel", product([0, 1], KERNELS))
def test_gradient_finiteness(random_state, kernel):
    """
    When x is the same as X_train, gradients might become undefined because
    they are divided by d(x, X_train).
    Check they are equal to numerical gradients at such points.
    """
    rng = np.random.RandomState(random_state)
    X = rng.randn(5).tolist()
    Y = [X]
    check_gradient_correctness(kernel, X, Y, 1e-6)
```